website/docs/docs/contributing/testing-a-new-adapter.md

```diff
@@ -0,0 +1,200 @@
+---
+title: "Testing a new adapter"
+id: "testing-a-new-adapter"
+---
+
+<VersionBlock lastVersion="1.0">
+
+The previous way to test dbt adapter functionality was via a packaged suite of tests, [`pytest-dbt-adapter`](https://github.com/dbt-labs/dbt-adapter-tests). We've since deprecated that approach in favor of the new testing framework outlined in this document.
+
+</VersionBlock>
+
+dbt-core offers a standard framework for running pre-built functional tests, and defining new ones.
+
+The core testing framework is built using `pytest`, a mature and standard library for testing Python projects. It assumes a basic level of familiarity with the foundational concepts in `pytest`, such as fixtures. If this is your first time using `pytest`, we recommend you read the documentation: https://docs.pytest.org/
+
+The [`tests` module](https://github.com/dbt-labs/dbt-core/tree/HEAD/core/dbt/tests) within `dbt-core` includes basic utilities for setting up `pytest` + `dbt`. These are built into `dbt-core`, and used by all functional tests. These also make it possible to write your own tests, using the same utilities.
```

Comment:
Realizing that the module contents here will actually be an dbt-core feature, for now undocumented, that is available to all project/package maintainers as well (if they care to write `pytest` cases)

Let's think about where the docs for this piece should live long-term, given that the intended audience is a bit wider than just creators/maintainers of adapter plugin

Reply:
We probably want some high level documentation of how to use the standard fixtures and structure a simple test.